File: server/src/main/java/com/senchacrm/domain/Person.java
package com.senchacrm.domain;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "people")
public class Person {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String firstName;
    private String lastName;
    private String title;
    private String company;
    private String phone;
    private String email;
    private String im;
    private String websites;
    private String info;
    private String photo;
    private Integer visited;
}

File: client/app/model/Person.js
Ext.define('SenchaCRM.model.Person', {
    extend: 'Ext.data.Model',

    requires: [
        'Ext.data.validator.Presence'
    ],

    fields: [
        {name: 'firstName', type: 'string'},
        {name: 'lastName', type: 'string'},
        {name: 'title', type: 'string'},
        {name: 'company', type: 'string'},
        {name: 'phone', type: 'string'},
        {name: 'email', type: 'string'},
        {name: 'im', type: 'string'},
        {name: 'websites', type: 'string'},
        {name: 'info', type: 'string'}
    ],

    validators: {
        firstName: 'presence',
        lastName: 'presence'
    },
    
    canSave: function () {
        return this.isValid();
    }

});
